PlayLoop/JumpHandler.js

```diff
@@ -3,8 +3,11 @@ class JumpHandler extends PlayMode {
     static jumpHandler() {
         const { player } = this;
 
-        if (Controller.jump && !player.collidingWithNpcId && !PauseHandler.justClosedPauseScreen ) {
-            if (player.swimming) {
+        if (Controller.jump) {
+            if (!PlayerInteractionHandler.willAllowJump() || PauseHandler.justClosedPauseScreen) {
+                // no jumping, sorry
+            }
+            else if (player.swimming) {
```

Comment:
why not just keep the condition as it was? 
```suggestion
        if (Controller.jump && !PlayerInteractionHandler.willAllowJump() || PauseHandler.justClosedPauseScreen) {
            if (player.swimming) {
```
If that's not possible, i would rethink how to make this condition. not a fan of empty IFs :P

Reply:
This is 99% just personal style choice for readability. Shorter conditional lines and nesting less deeply tends to make things more readable, in my experience. There are other approaches than a comment-only if, though.

Reply:
Oh, right! Actually one of the consequences of this (making the outer condition just be `Controller.jump` by itself) is that it prevents a bug where you can accidentally jump after passing interactable objects with the jump button already held down. It happens in the main branch right now and isn't too much of an issue there, but it can get weird in some edge cases with doors.